src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs

```diff
@@ -131,9 +131,10 @@ internal sealed class ClassEvaluator : MemberEvaluator {
                 return true;
             }
 
-            // Allow any members from typing module
-            // TODO handle typing module specialization better: https://github.com/microsoft/python-language-server/issues/1367
-            if (m is ILocatedMember l && l.DeclaringModule is TypingModule) {
+            // Allow extensions from specialized functions 
+            // We specialized type to be a function even though it is a class, so this allows extension of type
+            // TODO handle module specialization better: https://github.com/microsoft/python-language-server/issues/1367
+            if (m is IPythonType t && t.IsSpecialized) {
```

Comment:
Does changing this check from `typing` to the more general `IsSpecialized` negatively affect anything that was previously allowed from `typing`? I'm not sure how it works, but I think that `typing.ByteString` is a class and should work, but I don't think that it's "specialized". Maybe it's handled at a different point.

Reply:
`typing.ByteString` is treated the same as int, so it is a class, not specialized to a function. I've added another test with some of the typing module functions just to verify however.

Reply:
ByteString should be the same as `bytes`, but thanks for checking either way.

Reply:
Ah yes that's right.